atom/browser/lib/objects-registry.js

```diff
@@ -1,115 +1,110 @@
 const EventEmitter = require('events').EventEmitter;
+const util = require('util');
 const v8Util = process.atomBinding('v8_util');
 
-var extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };
-var hasProp = {}.hasOwnProperty;
+function ObjectsRegistry() {
+  ObjectsRegistry.super_.call(this);
 
-var ObjectsRegistry = (function(superClass) {
-  extend(ObjectsRegistry, superClass);
+  this.setMaxListeners(Number.MAX_VALUE);
+  this.nextId = 0;
 
-  function ObjectsRegistry() {
-    this.setMaxListeners(Number.MAX_VALUE);
-    this.nextId = 0;
+  // Stores all objects by ref-counting.
+  // (id) => {object, count}
+  this.storage = {};
 
-    // Stores all objects by ref-counting.
-    // (id) => {object, count}
-    this.storage = {};
+  // Stores the IDs of objects referenced by WebContents.
+  // (webContentsId) => {(id) => (count)}
+  this.owners = {};
+}
 
-    // Stores the IDs of objects referenced by WebContents.
-    // (webContentsId) => {(id) => (count)}
-    this.owners = {};
+util.inherits(ObjectsRegistry, EventEmitter);
+
+// Register a new object, the object would be kept referenced until you release
+// it explicitly.
+ObjectsRegistry.prototype.add = function(webContentsId, obj) {
+  var base, base1, id;
+  id = this.saveToStorage(obj);
+
+  // Remember the owner.
+  if ((base = this.owners)[webContentsId] == null) {
+    base[webContentsId] = {};
   }
+  if ((base1 = this.owners[webContentsId])[id] == null) {
```

Comment:
This is another line left over from #4065 that only appears in the diff here because of the outdenting.